### ISSUE TITLE
Test OMF with set -lx Fish changes

### ISSUE THREAD

**Issue by faho** [oh-my-fish/oh-my-fish#523]
It appears https://oh-my-fish-slack.herokuapp.com/ is broken since at least yesterday - I get an "Application Error".

Anyway, here's what I wanted to post:

>Hi all! We at fish-shell upstream would like to change how local-exported (set -lx) variables interact with functions. We think this is much better (in particular it enables begin; set -lx something something; somecommandorfunction; end to also work correctly with functions), but it can break some functions - those that expect to set a global var without defining it explicitly as global. Could y'all test [#4149](https://github.com/fish-shell/fish-shell/pull/4149) and see if it works with your functions?

**Comment by derekstavis** [comment]
Hey @faho, thanks for the reminder and your concern with community!! Unfortunately we had to remove the slack invitation broker because we had some issues with Fisherman author creating fake accounts to spam our Slack.

I will compile fish from #4149 and test if everything works with Oh My Fish!

Now, **my personal opinion**: I'd let this (and any other) breaking change for Fish 3.0. I suffered lots of breaks in last minor releases, ranging from colors to key bindings, and I wish Fish could follow semantic versioning! :+1:
